src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java

```diff
@@ -114,7 +112,7 @@ public RepositoryDelegatorFunction(
   private void setupRepositoryRoot(Path repoRoot) throws RepositoryFunctionException {
     try {
       FileSystemUtils.deleteTree(repoRoot);
-      FileSystemUtils.createDirectoryAndParents(repoRoot.getParentDirectory());
+      Preconditions.checkNotNull(repoRoot.getParentDirectory()).createDirectoryAndParents();
```

Comment:
nit: the precondition check is not necessary. If the argument is null, the NullPointerException will contain the exact same information as the exception thrown by the precondition check (this is not the only such place)

Reply:
Yes, of course you are right, but this is exactly the point of having Preconditions.checkNotNull() at all.
Personally, I also prefer to make a check like "env.valuesMissing()" vs checking some reference for null, so in this case the check "env.valuesMissing()" works in pair with "Preconditions.checkNotNull()" quite well in my opinion in term of expressiveness of code.
= You are saying "but this moment, this value is expected to be computed"